layer_test.go

```diff
@@ -503,6 +503,107 @@ func testLayer(t *testing.T, context spec.G, it spec.S) {
 			Expect(called).To(BeFalse())
 		})
 
+		it("does not call function with non-matching deprecation_date format", func() {
+			dependency = libpak.BuildpackDependency{
+				ID:      "test-id",
+				Name:    "test-name",
+				Version: "1.1.1",
+				URI:     fmt.Sprintf("%s/test-path", server.URL()),
+				SHA256:  "576dd8416de5619ea001d9662291d62444d1292a38e96956bc4651c01f14bca1",
+				Stacks:  []string{"test-stack"},
+				Licenses: []libpak.BuildpackDependencyLicense{
+					{
+						Type: "test-type",
+						URI:  "test-uri",
+					},
+				},
+				CPEs:            []string{"cpe:2.3:a:some:jre:11.0.2:*:*:*:*:*:*:*"},
+				PURL:            "pkg:generic/some-java11@11.0.2?arch=amd64",
+				DeprecationDate: dependency.DeprecationDate, // parsed as '2021-04-01 00:00:00 +0000 UTC'
```

Comment:
just looking via GH UI: 
```
dependency = libpak.BuildpackDependency{
  DeprecationDate: dependency.DeprecationDate
}
```
not sure what that means?!

Reply:
That tripped me up to. I believe it is referring to https://github.com/paketo-buildpacks/libpak/pull/261/files#diff-5f8f04e0f4cf91a97a2fee74038bd3131ae535e759a9696052843cad125b2192R392-R408 which is set in the before block.

Maybe it would be cleaner here to not redefine dependency in this test? so do `depWithDate := libpak...`

Reply:
ah true, I knew this had to do with Go variables visibility.
That's probably fine TBH, that'll teach me to review via the UI, missing important context